File: src/main/java/ninekothecat/catconomy/interfaces/ICatLogger.java
package ninekothecat.catconomy.interfaces;

import ninekothecat.catconomy.enums.TransactionResult;

import java.util.logging.Logger;

/**
 * The interface Cat logger.
 */
public interface ICatLogger {
    /**
     * logs a successful transaction.
     *
     * @param transaction the transaction
     */
    void success(ITransaction transaction);

    /**
     * logs a failed transaction
     *
     * @param transaction the transaction
     * @param result      the result
     */
    void fail(ITransaction transaction, TransactionResult result);

    /**
     * logs an error transaction
     *
     * @param transaction the transaction
     * @param result      the result
     * @param exception   the exception
     */
    void error(ITransaction transaction,TransactionResult result, Exception exception);
}

File: src/main/kotlin/ninekothecat/catconomy/interfaces/ICatLogger.kt
package ninekothecat.catconomy.interfaces

import ninekothecat.catplugincore.money.enums.TransactionResult
import ninekothecat.catplugincore.money.interfaces.ITransaction

/**
 * The interface Cat logger.
 */
interface ICatLogger {
    /**
     * logs a successful transaction.
     *
     * @param transaction the transaction
     */
    fun success(transaction: ITransaction)

    /**
     * logs a failed transaction
     *
     * @param transaction the transaction
     * @param result      the result
     */
    fun fail(transaction: ITransaction, result: TransactionResult)

    /**
     * logs an error transaction
     *
     * @param transaction the transaction
     * @param result      the result
     * @param exception   the exception
     */
    fun error(transaction: ITransaction, result: TransactionResult, exception: Exception?)
}